src/main/scala/strawman/collection/Factories.scala

```diff
@@ -47,6 +43,18 @@ object IterableFactory {
 
 }
 
+trait IterableFactoryWithBuilder[+CC[_]] extends IterableFactory[CC] {
```

Comment:
I think many of these names are way too long without being fully documenting (which would be the main reason to have long names).  We should aim for at most two words describing the type of collection (preferably one), and one word describing the concept except maybe two when it's an unusual case like Ordering.  I don't think we should worry about it now, though; let's get the functionality working first, and then worry about compact suggestive names.